### ISSUE TITLE
Check is json valid

### ISSUE THREAD

**Issue by egisxxegis** [daggaz/json-stream#35]
Hey all, 👋 
I am happy using json-stream and was wondering, what is the correct way to check json validity?
I tried doing it in the following way with `pytest` but it did not yield correct results:

```python
from io import StringIO

import pytest
import json_stream


def get_is_output_loadable(output: str):
    try:
        parsing = json_stream.load(StringIO(output))
        for data in parsing:
            pass
        return True
    except:
        return False


@pytest.mark.parametrize(
    ["output_str", "is_true_expected"],
    (
        ("{}", True),
        ("[]", True),
        ("[1, 2]", True),
        ('[{"type":"qqq"}, 9, [8, 9, 10]]', True),
        ('[{"name": "deep", "items": [1, {"name": "deeper", "numbers": null}]}]', True),
        ('["unterminated string]', False),
        ("[", False),
        ("{", False),
    ),
)
def test_is_output_loadable_correct(output_str: str, is_true_expected: bool):
    result = get_is_output_loadable(output_str)

    assert result == is_true_expected
```

**Comment by daggaz** [comment]
You are correct that these tests fail when they should not.

The examples all fall into the unterminated _something_ category.

The issue seems to be in the tokenizer, which is not recognising that reading `''` at that point in the stream is an error, and is instead just stopping the iteration.

Here are some examples of malformed JSON that _will_ be correctly identified by the tokenizer:
* `{]` (expecting })
* `"\p"` (invalid string escape)
* `truE` (bad true)

There are many more...

I'm away on a trip just now, so won't be able to look at this till I'm back.

Perhaps @smheidrich might be kind enough to look at it, as he'll need to port any changes to his rust library.

**Comment by daggaz** [comment]
@egisxxegis this has been resolved in json-stream 2.2.1

thanks for pointing this out!
